### ISSUE TITLE
TextIOWrapper: stream.tell() returns big 64 bits random number

### ISSUE THREAD

**Issue by mguinhos** [python/cpython#93077]
I was writing a lexer, and found this weird bug in python

**Python version:** Python 3.10.4
**Os:** Windows 11 21H2

```
...
Token.LeftParenthesis
char:  '"' tell:  102
None
char:  ')' tell:  18446744073709551746
-----> tell: 18446744073709551746
-----> tell: 18446744073709551745
Traceback (most recent call last):
  File "C:\Users\Usuário\Documents\Projetos\Bug\bug.py", line 11, in <module>
    for token in tokenize(open(fname)):
  File "C:\Users\Usuário\Documents\Projetos\Bug\tokenizer.py", line 85, in tokenize     
    yield scan_token(stream)
  File "C:\Users\Usuário\Documents\Projetos\Bug\tokenizer.py", line 58, in scan_token   
    raise SyntaxError(f'invalid token {stream.read(1)!r}')
SyntaxError: invalid token '\n'
```

i couldnt reproduce this bug with a example with less than 100 lines, but it is pretty organized

Obs:.
* MAX = 2**64 = 18446744073709551616
* If we subtract: MAX - 18446744073709551746 = -130

Here is the bug files:
[Bug.zip](https://github.com/python/cpython/files/8749965/Bug.zip)

**Comment by mguinhos** [comment]
If needed i will create a repo containing the bug src

**Comment by ronaldoussoren** [comment]
This probably is not a bug. TextIOBase.tell is documented to return an opaque value (see <https://docs.python.org/3/library/io.html#io.TextIOBase.tell>). The return value cannot be used to calculate differences between locations in a file.

**Comment by mguinhos** [comment]
I tested this in pypy3, and it doesnt have this bug.

The bug is that it jumps to this arbitrary number, before reaching the end of the file.

**Comment by mguinhos** [comment]
Also, i noticed that seek(offset) can take any number even if its out of file bounds.

Shouldnt this be fixed?

**Comment by Akuli** [comment]
Tiny reproducer. This prints 18446744073709551620.
```
import io

def take(stream):
    return stream.read(1)

def tokenize(stream):
    take(stream)
    take(stream)
    take(stream)
    print('-----> tell:', stream.tell())

with io.BytesIO(b'"")\r\nx\r\n') as f:
    tokenize(io.TextIOWrapper(f))
```

**Comment by Akuli** [comment]
And even tinier:

```python3
import io
stream = io.TextIOWrapper(io.BytesIO(b'"")\r\nx\r\n'))
stream.read(1)
stream.read(1)
stream.read(1)
print('-----> tell:', stream.tell())
```

**Comment by mguinhos** [comment]
Looks like it is happening with windows newlines?

**Comment by Akuli** [comment]
Indeed. Can't reproduce with just `\n`.

**Comment by Akuli** [comment]
> This probably is not a bug. TextIOBase.tell is documented to return an opaque value

This now makes sense, after looking at the C code. The return value of `.tell()` is calculated like this:

```c
static PyObject *
textiowrapper_build_cookie(cookie_type *cookie)
{
    unsigned char buffer[COOKIE_BUF_LEN];

    memcpy(buffer + OFF_START_POS, &cookie->start_pos, sizeof(cookie->start_pos));
    memcpy(buffer + OFF_DEC_FLAGS, &cookie->dec_flags, sizeof(cookie->dec_flags));
    memcpy(buffer + OFF_BYTES_TO_FEED, &cookie->bytes_to_feed, sizeof(cookie->bytes_to_feed));
    memcpy(buffer + OFF_CHARS_TO_SKIP, &cookie->chars_to_skip, sizeof(cookie->chars_to_skip));
    memcpy(buffer + OFF_NEED_EOF, &cookie->need_eof, sizeof(cookie->need_eof));

    return _PyLong_FromByteArray(buffer, sizeof(buffer),
                                 PY_LITTLE_ENDIAN, 0);
}
```

So it really doesn't represent the number of bytes since the start of the file. It's just a value that can be used to later `.seek()` back to the same place, built mostly from 4-byte and 8-byte chunks (`cookie->start_pos` is 8 bytes, `need_eof` is 1 byte, the rest are 4 bytes each).

Looking at the number as hex, it does look like it was built somewhat like this:

```
>>> hex(18446744073709551620)
'0x10000000000000004'
```

tl;dr: I now agree that this isn't a bug. I didn't know that `TextIOWrapper`'s file positions are not what you would expect them to be.

**Comment by mguinhos** [comment]
In the internal docs of tell()

```
Help on built-in function tell:

tell() method of _io.TextIOWrapper instance
    Return current stream position.
```

If it returns an opaque value, this description must be update

**Comment by mguinhos** [comment]
I will close this later, and open a new issue proposing behaviour changes (if possible).
